test/interval.jl

```diff
@@ -124,19 +124,47 @@ isinf(::TimeType) = false
         @test eltype(Interval{Float64}(1,2)) == Float64
     end
 
-    @testset "accessors" begin
-        for (a, b, _) in test_values
+    @testset "accessors - bounded intervals" begin
+        accessor_specific_vals = [
+            # test adding eps()
+            (-10.0, 10.0, nothing),
+            (-10.0, 10.0, eps(Float64)),
+
+            ('c', 'd', 2),
+            (Date(2004, 2, 13), Date(2020, 3, 13), Day(1)),
+        ]
+        for (a, b, unit) in append!(accessor_specific_vals, test_values)
             for (L, R) in BOUND_PERMUTATIONS
                 interval = Interval{L, R}(a, b)
 
                 @test first(interval) == a
                 @test last(interval) == b
+
+                # The value we compare to min/max depends on if the bound is open/closed.
+                # we can determine if a bound is open/closed by checking if the edge in the interval
+                mi = if a ∈ interval
+                    first(interval)
+                else
+                    v = first(interval)
+                    unit === nothing && eltype(interval) <: AbstractFloat ? nextfloat(v) : v + unit
+                end
+
+                ma = if b ∈ interval
+                    last(interval)
+                else
+                    v = last(interval)
+                    unit === nothing && eltype(interval) <: AbstractFloat ? prevfloat(v) : v - unit
+                end
+
+                @test minimum(interval; increment=unit) == mi
+                @test maximum(interval; increment=unit) == ma
```

Comment:
I think I'd combine these test with what you have below and call the testset "minimum/maximum"